src/auth/oauth2client.ts

```diff
@@ -793,6 +804,15 @@ export class OAuth2Client extends AuthClient {
       return {headers: this.addSharedMetadataHeaders(headers)};
     }
 
+    const downscopedCreds = await this.refreshHandler();
```

Comment:
Let's not use `downscopedCreds` as a name as this can be used in non-downscoping scenarios.

Reply:
Also since this is an optional property `refreshHandler`. You need to check it exists before running it.